magefiles/main.go

```diff
@@ -0,0 +1,149 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	"gopkg.in/yaml.v2"
+)
+
+// Mage default target.
+var Default = All
+
+// Content loaded from variables.yaml or variables.local.yaml
+// or file passed via VAGRANT_VARIABLES environment variable.
+var Settings map[string]string
+
+// Aliases based on original Makefile targets.
+// Use mage -h <target> to see available aliases.
+var Aliases = map[string]interface{}{
+	"1-build-binaries": Build,
+	"2-vagrant-up":     Run,
+	"3-smoke-test":     Test.Smoke,
+	"4-e2e-test":       Test.EndToEnd,
+}
+
+// All logs from Magefiles are prefixed with this tag (e.g. for easy search)
+const LogPrefix = "[swdt-mage] "
+
+// Run complete sequence of targets according to configuration in variables.yaml or variables.local.yaml.
+func All() error {
+	mg.SerialDeps(startup, Fetch, Run, Test.Smoke, Test.EndToEnd)
+
+	logTargetRunTime("All")
+	return nil
+}
+
+// Every public target must depend on startup.
+func startup() error {
+	startTime = time.Now()
+	log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile | log.LUTC | log.Lmsgprefix)
+	log.SetOutput(os.Stdout) // simplify use of tee
+	log.SetPrefix(LogPrefix)
+
+	// User may have already defined VAGRANT_VARIABLES environment variable
+	variablesFile := strings.TrimSpace(os.Getenv("VAGRANT_VARIABLES"))
+	if variablesFile != "" {
+		_, err := os.Stat(variablesFile)
+		if os.IsNotExist(err) {
+			variablesFile = ""
+		}
+	}
+	// Alternatively, user may have created local variables file
+	if variablesFile == "" {
+		_, err := os.Stat("variables.local.yaml")
+		if !os.IsNotExist(err) {
+			variablesFile = "variables.local.yaml"
+		}
+		if variablesFile != "" {
+			log.Printf("Setting environment variable VAGRANT_VARIABLES=%v", variablesFile)
+			os.Setenv("VAGRANT_VARIABLES", variablesFile)
```

Comment:
>  I think this will only set the environment variable for the currently running program, and it's not going to manipulate the caller's environment anyways.

Correct, the indention is to modify set the env var for current `mage` call in order to let `Vagrantfile` pick the `.local.yaml` via env variable, instead of the default `.yaml':

https://github.com/kubernetes-sigs/sig-windows-dev-tools/blob/56c162048e7d240882ca5eaeb92f550bd5f36e20/Vagrantfile#L6-L9